lib/source.c

```diff
@@ -109,6 +116,13 @@ tlog_source_read(struct tlog_source *source, struct tlog_pkt *pkt)
 
     assert(grc == TLOG_RC_OK || tlog_pkt_is_void(pkt));
     assert(tlog_source_is_valid(source));
+#ifndef NDEBUG
+    assert(!tlog_pkt_is_void(pkt));
```

Comment:
This is still using `assert`, which won't work. Think how you can make the below code check the condition when the packet is not "void", and not do any checking when it is, but without terminating the program.